src/cdk/tree/tree.ts

```diff
@@ -106,7 +106,11 @@ export class CdkTree<T> implements AfterContentChecked, CollectionViewer, OnDest
   @ViewChild(CdkTreeNodeOutlet, {static: true}) _nodeOutlet: CdkTreeNodeOutlet;
 
   /** The tree node template for the tree */
-  @ContentChildren(CdkTreeNodeDef) _nodeDefs: QueryList<CdkTreeNodeDef<T>>;
+  @ContentChildren(CdkTreeNodeDef, {
+    // We need to use `descendants: true`, because Ivy will no longer match
+    // indirect descendants if it's left as false.
+    descendants: true
```

Comment:
@jelbourn this could break if somebody were to nest `cdk-tree`, but I couldn't find any examples of it being supported. We support nesting tree nodes, but not the tree itself. We can support it, but it would need a bit of extra logic.

Reply:
IIRC we explicitly don't support nested trees (and tables) because of the trick we do for `mostRecent___Outlet`